### PR TITLE
sanitycheck: error on duplicate board identifiers

### PR DESCRIPTION
**Pull request by nashif**
2 boards had duplicate identifiers, fix those and add a check in sanitycheck to error when such issues are found.

### CODE REVIEW

File: scripts/sanity_chk/sanitylib.py [hunk 1]
```diff
@@ -2564,6 +2564,9 @@ def add_configurations(self):
                 try:
                     platform = Platform()
                     platform.load(file)
+                    if platform.name in [p.name for p in self.platforms]:
+                        logger.error(f"Duplicate platform {platform.name} in {file}")
+                        #raise Exception(f"Duplicate platform identifier {platform.name} found")
```
Comment (chrta) [inline scripts/sanity_chk/sanitylib.py:2569]: Is this comment intended?

**Comment by galak** [comment]
LGTM, same question as @chrta regarding the commented out code.

File: scripts/sanity_chk/sanitylib.py (hunk 1 above)
Comment (nashif) [reply scripts/sanity_chk/sanitylib.py]: fixed..

**Review by chrta — APPROVED** [review]

**Review by PerMac — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by jenmwms — APPROVED** [review]

**Comment by jenmwms** [comment]
I don't know if it is a blocking issue, but there are some `samples/` are getting cmake build error for the `lpcxpresso55s69_cpu1`, which I can reproduce locally. It works for `lpcxpresso55s69_cpu0`. Is that a problem?

**Comment by nashif** [comment]
> I don't know if it is a blocking issue, but there are some `samples/` are getting cmake build error for the `lpcxpresso55s69_cpu1`, which I can reproduce locally. It works for `lpcxpresso55s69_cpu0`. Is that a problem?

@jenmwms  yes, can you please open a bug an assign it to @MaureenHelm ? :-)

**Comment by jenmwms** [comment]
> @jenmwms yes, can you please open a bug an assign it to @MaureenHelm ? :-)

Thanks for suggesting! Please see https://github.com/zephyrproject-rtos/zephyr/issues/27880 capturing this.

**Comment by nashif** [comment]
lpcxpresso55s69_cpu1 needs some work still and it is preventing this PR from going in. Disabled sanitycheck for that board now (it never built in sanitycheck before anyways) until it is fixed but allowing the other fixes to go in so we catch issues that cause this to happen.

**Comment by abrodkin** [comment]
nSIM part is already addressed here https://github.com/zephyrproject-rtos/zephyr/pull/28379, so should be dropped from this one.

**Review by abrodkin — CHANGES_REQUESTED** [review]
Let's get rid of nsim part, as it's addressed here https://github.com/zephyrproject-rtos/zephyr/pull/28379 separately.

**Comment by nashif** [comment]
> Let's get rid of nsim part, as it's addressed here #28379 separately.

then you need to go approve that one.. :-)

**Comment by abrodkin** [comment]
@nashif done!

**Comment by nashif** [comment]
> @nashif done!

ok, patch removed and other PR merged.

**Review by galak — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Pull request merged**